external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala

```diff
@@ -38,6 +40,7 @@ private[kinesis] class KinesisInputDStream[T: ClassTag](
     val endpointUrl: String,
     val regionName: String,
     val initialPositionInStream: InitialPositionInStream,
+    val initialPositionInStreamTimestamp: Date,
```

Comment:
@budde - I had two approaches in mind while adding this functionality-
1. Additional parameter which can be set by an overloaded method in Builder.
2. Creating a new case class for wrapping initial position with an optional timestamp.

I went ahead with implementing the first one for backward compatibility such that users can use their same builders.